### PR TITLE
DOC: Add feature support by pypdf

### PR DESCRIPTION
**Pull request by j-t-1**

### CODE REVIEW

**Comment by stefan6419846** [comment]
What is the data based upon?

**Comment by j-t-1** [comment]
Content stream compression is provided by function `compress_content_streams`.
Cross-reference streams, do we support this? I will revert the tick.

**Comment by stefan6419846** [comment]
I will wait for @pubpub-zz to check if these statements are valid, having more knowledge on PDF files and specific stuff of *pypdf*.

**Comment by pubpub-zz** [comment]
Cross Reference streams are defined in §3.4.7of pdf 1.7 reference. They are tagged with XRefStm

It allows many objects to be stored within a stream which may improve compression.

pypdf can read them without problem. I can not remember any functions allowing to modify/create some.

File: docs/user/pdf-version-support.md [hunk 1]
```diff
@@ -18,14 +18,14 @@ all features of PDF 2.0.
 
 ## PDF Feature Support by pypdf
 
-| Feature                                 | PDF-Version | pypdf Support |
-| --------------------------------------- | ----------- | -------------- |
-| Transparent Graphics                    | 1.4         | ?              |
+| Feature                                 | PDF Version | pypdf Support  |
+| --------------------------------------- |:-----------:|:--------------:|
+| Transparent Graphics                    | 1.4         | ❓             |
```
Comment (pubpub-zz) [inline docs/user/pdf-version-support.md:23]: supported

File: docs/user/pdf-version-support.md [hunk 2]
```diff
@@ -18,14 +18,14 @@ all features of PDF 2.0.
...
 | CMaps                                   | 1.4         | ✅             |
-| Object Streams                          | 1.5         | ?              |
-| Cross-reference Streams                 | 1.5         | ?              |
-| Optional Content Groups (OCGs) - Layers | 1.5         | ?              |
-| Content Stream Compression              | 1.5         | ?              |
+| Object Streams                          | 1.5         | ❓             |
```
Comment (pubpub-zz) [inline docs/user/pdf-version-support.md:25]: supported

File: docs/user/pdf-version-support.md [hunk 3]
```diff
@@ -18,14 +18,14 @@ all features of PDF 2.0.
...
+| Cross-reference Streams                 | 1.5         | ✅             |
+| Optional Content Groups (OCGs) - Layers | 1.5         | ❓             |
```
Comment (pubpub-zz) [inline docs/user/pdf-version-support.md:27]: no dedicated function

**Comment by j-t-1** [comment]
Thanks @pubpub-zz. Content Stream Compression support is now the added supported feature.

**Comment by j-t-1** [comment]
I have reverted this so that the pypdf supported features is unchanged, so this is now just a cosmetic improvement.

**Comment by stefan6419846** [comment]
Why did you revert this? Apparently, the table has been outdated, thus I would recommend to update it. Just replacing the plain text symbols by emojis is not really something I would consider worth to change.

**Comment by stefan6419846** [comment]
It seems like the revert did too much - https://github.com/py-pdf/pypdf/pull/2738#pullrequestreview-2159481832 should be included.

File: docs/user/pdf-version-support.md [hunk 4]
```diff
@@ -18,14 +18,14 @@ all features of PDF 2.0.
...
-| --------------------------------------- | ----------- | -------------- |
-| Transparent Graphics                    | 1.4         | ?              |
+| Feature                                 | PDF Version | pypdf Support  |
+| --------------------------------------- |:-----------:|:--------------:|
 | CMaps                                   | 1.4         | ✅             |
-| Object Streams                          | 1.5         | ?              |
-| Cross-reference Streams                 | 1.5         | ?              |
-| Optional Content Groups (OCGs) - Layers | 1.5         | ?              |
-| Content Stream Compression              | 1.5         | ?              |
+| Transparent Graphics                    | 1.4         | ❓             |
```
Comment (stefan6419846) [inline docs/user/pdf-version-support.md:24]:
```suggestion
| Transparent Graphics                    | 1.4         | ✅             |
```

File: docs/user/pdf-version-support.md [hunk 5]
```diff
@@ -18,14 +18,14 @@ all features of PDF 2.0.
...
+| Content Stream Compression              | 1.5         | ✅             |
+| Cross-reference Streams                 | 1.5         | ✅             |
```
Comment (stefan6419846) [inline docs/user/pdf-version-support.md:26]:
```suggestion
| Cross-reference Streams                 | 1.5         | ❓             |
```

File: docs/user/pdf-version-support.md [hunk 6]
```diff
@@ -18,14 +18,14 @@ all features of PDF 2.0.
...
+| Object Streams                          | 1.5         | ❓             |
```
Comment (stefan6419846) [inline docs/user/pdf-version-support.md:27]:
```suggestion
| Object Streams                          | 1.5         | ✅             |
```

**Review by stefan6419846 — APPROVED** [review]

**Pull request merged**